autolab/Autolab
# Git submission UI

## Description
This PR depends on #1320 and creates the submission form UI for assessments that have Git submissions enabled.

<img width="1011" alt="Screen Shot 2021-04-12 at 8 08 31 PM" src="https://user-images.githubusercontent.com/13857201/114477791-dc7fe380-9bca-11eb-945b-eb90f016bf08.png">

## Motivation and Context
This is part of the new Git submission feature that is currently in progress.

## How Has This Been Tested?

1. Follow steps 1-4 in #1320 to create a new assignment on Github Classroom (should be named {assignment name}-{student name})
2. Create a personal [Github access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
3. Go to course settings to fill in the Git submission fields (Git Access Key, Git Username, Git Classroom Name)
4. Go to Autograder settings for an assessment, click the Enable git submissions checkbox, and fill in the Git Assignment Name
5. The assessment submission form should now have Git Username and Commit Hash fields instead of a file upload box
6. Fill in the fields, check the academic integrity box, and then hit the submit button, which should be enabled only once the fields are filled in
7. The submission should appear, and when clicking the magnifying class, the content should be from the correct commit

<img width="1440" alt="Screen Shot 2021-04-12 at 8 27 36 PM" src="https://user-images.githubusercontent.com/13857201/114479002-86606f80-9bcd-11eb-818b-86a4cb2d8097.png">

## Types of changes


## Checklist:


## Other issues / help required




If unsure, feel free to submit first and we'll help you along.
